src/otx/algo/detection/heads/yolo_head.py

```diff
@@ -0,0 +1,729 @@
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward function."""
+        return self.implicit * x
+
+
+class SingleHeadDetection(nn.Module):
+    """A single YOLO Detection head for YOLOv9 detection models.
+
+    Args:
+        in_channels (tuple[int, int]): Number of input channels.
+        num_classes (int): Number of classes.
+        reg_max (int): Maximum number of anchor regions.
+        use_group (bool): Whether to use group convolution.
+    """
+
+    def __init__(
+        self,
+        in_channels: tuple[int, int],
+        num_classes: int,
+        *,
+        reg_max: int = 16,
+        use_group: bool = True,
+    ) -> None:
+        super().__init__()
+
+        groups = 4 if use_group else 1
+        anchor_channels = 4 * reg_max
+
+        first_neck, first_channels = in_channels
+        anchor_neck = max(round_up(first_neck // 4, groups), anchor_channels, reg_max)
+        class_neck = max(first_neck, min(num_classes * 2, 128))
+
+        self.anchor_conv = nn.Sequential(
+            Conv2dModule(
+                first_channels,
+                anchor_neck,
+                3,
+                padding=auto_pad(3),
+                normalization=nn.BatchNorm2d(anchor_neck, eps=1e-3, momentum=3e-2),
+                activation=nn.SiLU(inplace=True),
+            ),
+            Conv2dModule(
+                anchor_neck,
+                anchor_neck,
+                3,
+                padding=auto_pad(3),
+                groups=groups,
+                normalization=nn.BatchNorm2d(anchor_neck, eps=1e-3, momentum=3e-2),
+                activation=nn.SiLU(inplace=True),
+            ),
+            nn.Conv2d(anchor_neck, anchor_channels, 1, groups=groups),
+        )
+        self.class_conv = nn.Sequential(
+            Conv2dModule(
+                first_channels,
+                class_neck,
+                3,
+                padding=auto_pad(3),
+                normalization=nn.BatchNorm2d(class_neck, eps=1e-3, momentum=3e-2),
+                activation=nn.SiLU(inplace=True),
+            ),
+            Conv2dModule(
+                class_neck,
+                class_neck,
+                3,
+                padding=auto_pad(3),
+                normalization=nn.BatchNorm2d(class_neck, eps=1e-3, momentum=3e-2),
+                activation=nn.SiLU(inplace=True),
+            ),
+            nn.Conv2d(class_neck, num_classes, 1),
+        )
+
+        self.anc2vec = Anchor2Vec(reg_max=reg_max)
+
+        self.anchor_conv[-1].bias.data.fill_(1.0)
+        self.class_conv[-1].bias.data.fill_(-10)  # TODO (author): math.log(5 * 4 ** idx / 80 ** 3)
+
+    def forward(self, x: Tensor) -> tuple[Tensor, Tensor, Tensor]:
+        """Forward function."""
+        anchor_x = self.anchor_conv(x)
+        class_x = self.class_conv(x)
+        anchor_x, vector_x = self.anc2vec(anchor_x)
+        return class_x, anchor_x, vector_x
+
+
+class ISingleHeadDetection(nn.Module):
```

Comment:
The name of this class is a bit strange for me. If it is used for V7, maybe we could rename these two classes as SingleHeadDetectionV9 and SingleHeadDetectionV7...

Reply:
Done at 8406227